electionleaflets/apps/leaflets/views.py

```diff
@@ -230,23 +240,22 @@ def done(self, form_list, **kwargs):
                     )
                     leaflet.delete()
                     return redirect(reverse("upload_leaflet"))
-                
-                uploaded_images = json.loads(images_text)
-                bucket = self.storage.file_storage.bucket
+
                 for file_path in uploaded_images:
                     file_name = file_path.split("/")[-1]
                     file_name = file_name.replace(" ", "-")
-                    copy_source = {
-                        "Bucket": bucket.name,
-                        "Key": file_path,
-                    }
+                    
+                    storage_backend = getattr(settings, 'DEFAULT_FILE_STORAGE')
+                    bucket = self.storage.file_storage.bucket
+
                     new_file_name = f"leaflets/{leaflet.pk}/{file_name}"
-                    moved_file = bucket.Object(new_file_name)
-                    moved_file.copy(copy_source)
 
-                    new_file_name_raw = f"raw_{new_file_name}"
-                    moved_file_raw = bucket.Object(new_file_name_raw)
-                    moved_file_raw.copy(copy_source)
+                    if storage_backend != 'django.core.files.storage.FileSystemStorage':
```

Comment:
Can you invert this logic to test if we're using the backend we're targetting, rather than _not_ using a different backend? Partly because I find `if thing` is easier to read than `if not other_thing`, but also because there's nothing to say that there will only ever be two storages backends, so we can't rely on 'not file system' meaning 's3'.

Reply:
Done with https://github.com/DemocracyClub/electionleaflets/pull/259/commits/d9e731a7ecfb63a927ee40f4e2dce14d6d237ad3

Reply:
> Done with [d9e731a](https://github.com/DemocracyClub/electionleaflets/commit/d9e731a7ecfb63a927ee40f4e2dce14d6d237ad3)

Sorry, this still has the same problem: by putting all the code we want to run on an S3 backend in the `else`, we're just catching that we're _not_ using `FileSystemStorage` (but could be using any number of other backends). 

Can you make it explicit? `if backend == "storages.backends.s3boto3.S3Boto3Storage"...` do the thing we want to do on that backend?

Reply:
Sorry! Rushed that fix. Now hopefully done with https://github.com/DemocracyClub/electionleaflets/pull/259/commits/b4a2f037f65741134d58d66888465689cb722677